lib/pure/os.nim

```diff
@@ -3217,6 +3184,52 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns `true` if both pathname arguments refer to files with identical binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ##
+  ## .. code-block:: nim
+  ##   echo sameFileContent("file0.txt", "file1.txt", checkSize = true, bufferSize = 999)
+  ##
+  ## See also:
+  ## * `sameFile proc <#sameFile,string,string>`_
+  var a, b: File
+  var mustRead = true
+  var bufA = alloc(bufferSize)
+  var bufB = alloc(bufferSize)
+  try:  # readBuffer or open may or may not raise IOError.
+    if not open(a, path1):
+      raise newException(IOError, "Can not open file: $1" % [path1])
```

Comment:
I now realize that before PR no exception would be raised, which seems buggy, eg:
`sameFileContent("nonexistant", "nonexistant")` would silently return false.
I prefer API's that raise rather than silently allow invalid inputs but given that this was previous behavior, we should let others chime in here. One possibility is a `checkArgs` but it also adds a param.

Reply:
I am copying the solution used on `os` "checkDir".